TESTING.md

```diff
@@ -0,0 +1,32 @@
+Running tests
+=============
+
+The Datadog Go API client has 2 sorts of tests: regular tests and BDD tests.
+Both are using cassettes to record HTTP interactions, allowing to run the tests without
+talking to the API. We also store the test time to be able to freeze.
+
+You can control the behavior with the `RECORD` environment variable:
+ - `RECORD=false`, the default value, means replaying HTTP requests from recording.
+ - `RECORD=true` creates or updates recordings. This will need valid credentials in `DD_TEST_CLIENT_API_KEY`
+    and `DD_TEST_CLIENT_APP_KEY`.
+ - `RECORD=none` ignores recordings. This will also runs tests that we call `integration-only`, ie
+    tests that we don't record for security reasons. It also needs valid credentials.
+
+Cassettes and freeze files are stored in `tests/api/$VERSION/datadog/cassettes/`, with one file per tests.
```

Comment:
In order to make it more future proof can we say something like:

```
Cassettes and freeze files are stored in `cassettes` directory in each test package with one file per tests (e.g. `tests/api/v1/datadog/cassettes/`).
```
Since cassettes for scenarios will be in `tests/scenarios/cassettes/TestScenario/$VERSION/...` after #1145 is merge.